quic/s2n-quic-tls/src/server.rs

```diff
@@ -206,6 +206,11 @@ impl Builder {
             params: Default::default(),
         })
     }
+
+    /// Returns the underlying [`Config`](s2n_tls::config::Config).
+    pub fn into_config(self) -> Result<Config, Error> {
```

Comment:
I'd rather this be a `impl Into<Config> for Server`

Reply:
From Server seems like over kill. I like the From Builder more. Is there a reason why it should be from Server?

```
impl From<Builder> for Result<Config, Error> {
     fn from(server: Builder) -> Self {
         server.config.build()
    }
}

impl<L: ConfigLoader> From<Server<L>> for Config {
     fn from(mut server: Server<L>) -> Self {
         server.load(crate::ConnectionContext { server_name: None })
     }
}

```

Reply:
I don't like conflating the builder here. It's a _server_ builder, not a config builder. But converting a built server _into_ a config seems reasonable.

Reply:
Implementing into for Result is pretty rare. Usually you'd do try_into.

Reply:
Yea associating with the Server conceptually makes more sense.

The syntax does become more ugly with the `map` and personally I dont like `into()` since its not always possible to read the code and know the types.

```
let config = s2n_quic::provider::tls::s2n_tls::Server::builder()
                     .with_certificate(cert, key)?
                     .build()
                     .map(|s| s.into());
```

Reply:
```rust
let config = s2n_quic::provider::tls::s2n_tls::Server::builder()
                     .with_certificate(cert, key)?
                     .build()?
                     .into();
```

Reply:
duh